docs/pipelines.md

```diff
@@ -465,6 +465,22 @@ There are a lot of scenarios where `WhenExpressions` can be really useful. Some
 - Checking if the name of a CI job matches
 - Checking if an optional Workspace has been provided
 
+#### Guarding a `Task` and its dependent `Tasks`
+
+When  `WhenExpressions` evaluate to `False`, the `Task` and its branch (of dependent `Tasks`) will be skipped by
```

Comment:
i think it would be good to define (or link to a definition elsewhere) of what "dependent tasks" means

There are also some docs on "conditions" that look like they might be trying to explain this - might be worth borrowing and/or updating those as well: https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#guard-task-execution-using-conditions